cookbooks/aws-parallelcluster-slurm/files/default/head_node_checks/common/constants.py

```diff
@@ -12,7 +12,7 @@
 
 # BOTO
 BOTO_CONFIG = Config(retries={"max_attempts": 60})
-BOTO_PAGINATION_CONFIG = {"PageSize": 500}
+BOTO_PAGINATION_CONFIG = {"PageSize": 100}
```

Comment:
To address scaling issues caused by DFSMv2, we are decreasing the pagination size from 500 to 100 items, making both EC2 calls and DDB calls retrieve at most 100 items per request. This is fine but can be improved post release. In fact, DDB request BatchGetItem supports at most 100 elements, but the EC2 DescribeInstances can support 500. So with this change we increase by 5x the number of EC2 requests.

We want to improve this approach by making the EC2 DescribeInstances retrieve 500 items per request and execute the DDB request in batch of 100 items each. In this way we would take the most out of each request, minimizing the number of request and the consequent risk of throttling.